src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -0,0 +1,5071 @@
+            {
+                if (contents.Length > destination.Length)
+                {
+                    bytesWritten = 0;
+                    unusedBitCount = 0;
+                    return false;
+                }
+
+                contents.Span.CopyTo(destination);
+                bytesWritten = contents.Length;
+                // contents doesn't include the unusedBitCount value, so add one byte for that.
+                _data = _data.Slice(headerLength + contents.Length + 1);
+                return true;
+            }
+
+            // Expected vs actual tag was already checked in TryGetBitStringBytes.
+            // Either constructed, or a BER payload with "unused" bits not set to 0.
+            (Asn1Tag tag, int? length) = ReadTagAndLength(out headerLength);
+
+            if (!tag.IsConstructed)
+            {
+                Debug.Assert(_ruleSet == AsnEncodingRules.BER);
+
+                return TryCopyPrimitiveBitStringValue(
+                    Slice(_data, headerLength, length),
+                    destination,
+                    true,
+                    false,
+                    _ruleSet,
+                    out unusedBitCount,
+                    out bytesWritten);
+            }
+
+            bool read = TryCopyConstructedBitStringValue(
+                Slice(_data, headerLength, length),
+                destination,
+                _ruleSet,
+                length == null,
+                out unusedBitCount,
+                out int bytesRead,
+                out bytesWritten);
+
+            if (read)
+            {
+                _data = _data.Slice(headerLength + bytesRead);
+            }
+
+            return read;
+        }
+
+        public TFlagsEnum GetNamedBitListValue<TFlagsEnum>() where TFlagsEnum : struct =>
+            GetNamedBitListValue<TFlagsEnum>(Asn1Tag.PrimitiveBitString);
+
+        public TFlagsEnum GetNamedBitListValue<TFlagsEnum>(Asn1Tag expectedTag) where TFlagsEnum : struct
+        {
+            Type tFlagsEnum = typeof(TFlagsEnum);
+
+            return (TFlagsEnum)Enum.ToObject(tFlagsEnum, GetNamedBitListValue(expectedTag, tFlagsEnum));
+        }
+
+        public Enum GetNamedBitListValue(Type tFlagsEnum) =>
+            GetNamedBitListValue(Asn1Tag.PrimitiveBitString, tFlagsEnum);
+
+        public Enum GetNamedBitListValue(Asn1Tag expectedTag, Type tFlagsEnum)
+        {
+            // This will throw an ArgumentException if TEnum isn't an enum type,
+            // so we don't need to validate it.
+            Type backingType = tFlagsEnum.GetEnumUnderlyingType();
+
+            if (!tFlagsEnum.IsDefined(typeof(FlagsAttribute), false))
+            {
+                throw new ArgumentException(
+                    SR.Cryptography_Asn_NamedBitListRequiresFlagsEnum,
+                    nameof(tFlagsEnum));
+            }
+
+            int sizeLimit = Marshal.SizeOf(backingType);
+            Span<byte> stackSpan = stackalloc byte[1 + sizeLimit];
```

Comment:
Why does this add 1?

Reply:
It probably started off as a place for a padding byte for `ulong`.  But I just removed it and all the tests still pass, including the `ulong` test which sets both "0" and "63" (0b100...001)